plugins/source/postgresql/client/list_tables.go

```diff
@@ -65,20 +65,26 @@ func (c *Client) listTables(ctx context.Context) (schema.Tables, error) {
 		return nil, err
 	}
 	defer rows.Close()
+
+	tableMap := make(map[string]*schema.Table)
+
 	for rows.Next() {
 		var ordinalPosition int
 		var tableName, columnName, columnType, pkName string
 		var isPrimaryKey, notNull, isUnique bool
 		if err := rows.Scan(&ordinalPosition, &tableName, &columnName, &columnType, &isPrimaryKey, &notNull, &isUnique, &pkName); err != nil {
 			return nil, err
 		}
-		if ordinalPosition == 1 {
-			tables = append(tables, &schema.Table{
+		table := tableMap[tableName]
+		if table == nil {
+			table = &schema.Table{
 				Name:    tableName,
 				Columns: make([]schema.Column, 0),
-			})
+			}
+			tableMap[tableName] = table
+			tables = append(tables, table)
 		}
-		table := tables[len(tables)-1]
```

Comment:
This line caused problems for a user in discord, can't be sure why (resultset isn't empty but ordinalPosition didn't start at 1...)